src/dawn/IIR/IIRSerializer.cpp

```diff
@@ -0,0 +1,623 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+#include "dawn/IIR/IIRSerializer.h"
+#include "dawn/IIR/IIR.pb.h"
+#include "dawn/IIR/IIRNodeIterator.h"
+#include "dawn/IIR/MultiStage.h"
+#include "dawn/IIR/StatementAccessesPair.h"
+#include "dawn/IIR/StencilInstantiation.h"
+#include "dawn/Optimizer/PassComputeStageExtents.h"
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Support/ASTSerialier.h"
+#include <fstream>
+#include <google/protobuf/util/json_util.h>
+#include <list>
+#include <stack>
+#include <tuple>
+#include <utility>
+
+using namespace dawn;
+
+static void setAccesses(proto::iir::Acesses* protoAcesses,
+                        const std::shared_ptr<iir::Accesses>& accesses) {
+  auto protoReadAccesses = protoAcesses->mutable_readaccess();
+  for(auto IDExtentsPair : accesses->getReadAccesses()) {
+    proto::iir::Extents protoExtents;
+    iir::Extents e = IDExtentsPair.second;
+    for(auto extent : e.getExtents()) {
+      auto protoExtent = protoExtents.add_extents();
+      protoExtent->set_minus(extent.Minus);
+      protoExtent->set_plus(extent.Plus);
+    }
+    protoReadAccesses->insert({IDExtentsPair.first, protoExtents});
```

Comment:
There is something weird in the accesses. If I deserialized the data you gave me, I get the following: 
for example for the statement: 
```          float hori_adv_dr_v=eddlat * adv_dr_1d_v_i + eddlon * adv_dr_1d_v_j```
I get
```
          w: __local_hori_adv_dr_v_103{0,0,0,0,0,0}
          r: __local_hori_adv_dr_v_103{0,0,0,0,0,0}
          r: __local_hori_adv_dr_v_103{0,0,0,0,0,0}
          r: __local_hori_adv_dr_v_103{0,0,0,0,0,0}
          r: __local_hori_adv_dr_v_103{0,0,0,0,0,0}
```

Reply:
I'll look at it